nym-api/src/coconut/keys/persistence.rs

```diff
@@ -3,17 +3,17 @@
 
 use crate::coconut::keys::KeyPairWithEpoch;
 use crate::coconut::state::bandwidth_credential_params;
-use nym_coconut::{CoconutError, KeyPair, SecretKey};
 use nym_coconut_dkg_common::types::EpochId;
+use nym_compact_ecash::{error::CompactEcashError, scheme::keygen::SecretKeyAuth, KeyPairAuth};
 use nym_pemstore::traits::PemStorableKey;
 use std::mem;
 
 impl PemStorableKey for KeyPairWithEpoch {
     // that's not the best error for this, but it felt like an overkill to define a dedicated struct just for this purpose
-    type Error = CoconutError;
+    type Error = CompactEcashError;
 
     fn pem_type() -> &'static str {
-        "COCONUT KEY WITH EPOCH"
+        "COCONUT KEY WITH EPOCH" // avoid the invalidation of already present key
```

Comment:
I don't think we want to be hacking our compatibility with coconut keys this way. 
I'd say, define a brand new pem type with brand new serialisation (in case we changed something in the serde crypto code in the previous PR).
Then, on nym-api startup (or coconut module creation/key load) see if you have old coconut keys. if so, convert them into ecach keys via `From` trait